lib/src/main/java/graphql/nadel/engine/transform/NadelTypeRenameResultTransform.kt

```diff
@@ -77,13 +77,16 @@ internal class NadelTypeRenameResultTransform : NadelTransform<State> {
                 underlyingTypeName = underlyingTypeName,
             )
 
-            val typeName: String = if (overallField.objectTypeNames.contains(overallTypeName)) {
+            val typeName: String = if (
+                !executionContext.hints.sharedTypeRenames(service) ||
```

Comment:
Sorry idk if my brain is fried but I've been staring at this if statement for a 5 mins trying to figure it out and I can't lol.

Can we change it to something clearer like

```
val typeName: String = if (executionContext.hints.sharedTypeRenames(service)) {
    if (overallField.objectTypeNames.contains(overallTypeName)) {
        overallTypeName
    } else {
        overallField.objectTypeNames.singleOrNull {
            executionBlueprint.getRename(it)?.underlyingName == underlyingTypeName
        } ?: overallTypeName
    }
} else {
    overallTypeName
}
```

Reply:
thanks, done